plugins/org.jkiss.dbeaver.core/src/org/jkiss/dbeaver/ui/e4/DBeaverCTabFolderRenderer.java

```diff
@@ -128,6 +127,27 @@ private static IEditorInput getEditorInput(@NotNull MPart part) {
         return null;
     }
 
+    @Nullable
+    private static Color getConnectionColor(@NotNull IEditorPart editorPart) {
```

Comment:
It is better to use DBPDataSourceContainerProvider but DBPContextProvider (as DBPContextProvider works only with connected databases)